From 8ff413c28358d781e1b9416c69d0fdab79cfdd48 Mon Sep 17 00:00:00 2001
From: Saikiran <bjsaikiran@gmail.com>
Date: Mon, 26 Jan 2026 17:23:21 +0530
Subject: [PATCH 5/6] arm64: dts: qcom: x1e80100-yoga-slim7x: Add off-on-delay
 to camera regulators

The AVDD (2.8V), DVDD (1.2V), and DOVDD (1.8V) camera regulators on the
Lenovo Yoga Slim 7x rely on passive discharge, which takes approximately
2.3 seconds to reach a safe low-voltage state.

If the camera sensor is powered on before the rails have fully discharged,
the sensor enters a brownout state and fails to initialize, requiring a
full system reboot to recover.

Add 'regulator-off-on-delay-us' properties to enforce a 2.3s delay after
power-off, preventing brownouts during rapid power cycling.

Signed-off-by: Saikiran <bjsaikiran@gmail.com>
---
 arch/arm64/boot/dts/qcom/x1e80100-lenovo-yoga-slim7x.dts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/x1e80100-lenovo-yoga-slim7x.dts b/arch/arm64/boot/dts/qcom/x1e80100-lenovo-yoga-slim7x.dts
index 8f1047ff2b0e..4cbc151ea72a 100644
--- a/arch/arm64/boot/dts/qcom/x1e80100-lenovo-yoga-slim7x.dts
+++ b/arch/arm64/boot/dts/qcom/x1e80100-lenovo-yoga-slim7x.dts
@@ -573,6 +573,7 @@ vreg_l7b_2p8: ldo7 {
 			regulator-min-microvolt = <2800000>;
 			regulator-max-microvolt = <2800000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
+			regulator-off-on-delay-us = <2300000>;
 		};
 
 		vreg_l8b_3p0: ldo8 {
@@ -832,6 +833,7 @@ vreg_l1m_1p2: ldo1 {
 			regulator-min-microvolt = <1200000>;
 			regulator-max-microvolt = <1260000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
+			regulator-off-on-delay-us = <2300000>;
 		};
 
 		vreg_l2m_1p2: ldo2 {
@@ -846,6 +848,7 @@ vreg_l3m_1p8: ldo3 {
 			regulator-min-microvolt = <1800000>;
 			regulator-max-microvolt = <1900000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
+			regulator-off-on-delay-us = <2300000>;
 		};
 
 		vreg_l4m_1p8: ldo4 {
-- 
2.51.0

